### DISCUSSION TITLE
Netbox backup

### DISCUSSION THREAD

**Discussion by ccraluca** [netbox-community/netbox#6534] [Q&A] [answered]
Hi there,

I'm new to netbox. I managed to install it on Ubuntu 20.04 and enter some test information.
Now I want to make a backup; is there any tutorial with more detailed steps and explanations? I want to understand exactly what's happening with database and how to restore it...

Thank you!

**Comment by candlerb** [comment, marked as answer]
> Now I want to make a backup; is there any tutorial with more detailed steps and explanations?

Yes. https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/

**Reply by ccraluca** [reply]
A database backed up from an older version of netbox can be restored in a newer version? I'm trying to do this and I receive some errors..

netbox_1         | django.db.utils.ProgrammingError: column "custom_field_data" of relation "dcim_devicerole" already exists

**Reply by candlerb** [reply]
Yes it can.  It has to be restored as a completely new database - i.e. on the target delete (or rename) any existing 'netbox' database first.  And then once you've restored it, you have to run the `./upgrade.sh` script to apply database migrations to modify it appropriately.

**Reply by ccraluca** [reply]
With netbox-docker is same procedure?

**Reply by candlerb** [reply]
I believe the docker container will [run the database migration](https://github.com/netbox-community/netbox-docker/blob/release/docker/docker-entrypoint.sh#L18) when it starts up.  You'll need to check what happens to media files, scripts, reports etc.  AFAICS, docker-compose.yml uses a docker volume for the [media files](https://github.com/netbox-community/netbox-docker/blob/release/docker-compose.yml#L18) but not the others.
